File: 프로그래머스/0/120890. 가까운 수/가까운 수.cpp
#include <bits/stdc++.h>

using namespace std;

int solution(vector<int> array, int n) {
    int answer = 0;
    int temp = INT_MAX;
    for(int i : array) {
        int _diff = abs(i - n);
        if(_diff < temp) {
            temp = _diff;
            answer = i;
        } else if(_diff == temp) {
            if(answer > i) answer = i;
        }
    }
    return answer;
}

File: 프로그래머스/0/120890. 가까운 수/가까운 수.swift
import Foundation

func solution(_ array:[Int], _ n:Int) -> Int {
    var diff = Int.max, answer = 0
    array.forEach {
        if diff > abs($0 - n) {
            diff = abs($0 - n)
            answer = $0
        } else if diff == abs($0 - n) && answer > $0 {
            answer = $0
        }
    }
    return answer
}